packages/dds/tree/src/core/tree/anchorSet.ts

```diff
@@ -564,19 +573,49 @@ export class AnchorSet implements ISubscribable<AnchorSetRootEvents> {
 			}
 		};
 
+		const pathVisitors: Map<PathNode, PathVisitor[]> = new Map();
+
 		const visitor: DeltaVisitor = {
 			onDelete: (start: number, count: number): void => {
 				assert(parentField !== undefined, 0x3a7 /* Must be in a field to delete */);
 				maybeWithNode(
-					(p) => p.events.emit("childrenChanging", p),
+					(p) => {
+						p.events.emit("childrenChanging", p);
+						if (parentField !== undefined) {
```

Comment:
This is unconditionally true (though typescript probably can't tell).

Also if you put this inside maybeWithNode, it won't get run if there is no anchor in this subtree, and one of the main points of this API is to have it run on subtrees without anchors!

I pushed fixed for this and some improved docs to and a few other changes to https://github.com/CraigMacomber/FluidFramework/tree/external-delta-visitor . I was unable to push them to your branch or created A PR into your branch with the changes.

Reply:
>  I was unable to push them to your branch or created A PR into your branch with the changes.

Probably a permission problem. Added your user to the repo. Please accept the invite

Reply:
Thank you for the contribution. Merged your branch in the current PR.